src/symbolic/symbolic_choice.ml

```diff
@@ -366,7 +366,7 @@ let get_model_or_stop symbol =
     | Some v -> return v
   end
 
-let select (cond : Symbolic_value.vbool) =
+let select_inner ?(explore_first = true) (cond : Symbolic_value.vbool) =
```

Comment:
Could you use `~explore_first` instead ? I don't really like optional arguments TBH, and here the function is used in only two places so it's not that tedious